Support error cause (#27837)
DataDog/documentation: content/en/real_user_monitoring/browser/collecting_browser_errors.md

| `error.type`    | string | The error type (or error code in some cases).                     |
| `error.message` | string | A concise, human-readable, one-line message explaining the event. |
| `error.stack`   | string | The stack trace or complementary information about the error.     |

> | `error.causes`  | [Array][12] | An optional nested error providing additional context. Use this attribute to display errors separately and enhance formatting. [More info from the MDN documentation.][13]   |

Reviewer: **💬 suggestion:** ‏
```suggestion
| `error.causes`  | [Array][12] | An optional list of errors providing additional context. This attribute is used to display errors separately and enhance formatting. [More info from the MDN documentation.][13]   |
```

|-----------------|--------|-------------------------------------------------------------------|
| `error.type`    | string | The error type (or error code in some cases).                     |

### Error Cause

> Using the `error.cause` attribute allows you to chain errors more clearly. With this property, each error in the chain is displayed separately, ensuring that error messages and stack traces remain distinct with a better formating.

Reviewer 2: ```suggestion
Using the `error.cause` attribute allows you to chain errors more clearly. With this property, each error in the chain is displayed separately, ensuring that error messages and stack traces remain distinct with a better formating.
```
```suggestion
Use the `error.cause` attribute to chain errors clearly. This property displays each error in the chain separately, ensuring that error messages and stack traces remain distinct with improved formatting.
```

| Attribute      | Type  | Description                                                                                          |
|----------------|-------|------------------------------------------------------------------------------------------------------|

> | `error.cause`  | Error | An optional nested error providing additional context. Use this attribute to display errors separately and enhance formatting. |

Reviewer: **💬 suggestion:** 
* The attribute should be named `error.causes`
* This row should be listed in the "Error attributes" table above. A specific section might not be needed, we can just link to MDN documentation in the attribute description.
* The `Type` should reflect the type of this attribute [in the RUM event](https://github.com/DataDog/rum-events-format/blob/master/lib/esm/generated/rum.d.ts#L185-L203). As it is not a simple type, maybe we could just write `Array` but put a link to [the RUM event format](https://github.com/DataDog/rum-events-format/blob/69147431d689b3e59bff87e15bb0088a9bb319a9/lib/esm/generated/rum.d.ts#L185-L203)?

> | `error.cause`  | Error | An optional nested error providing additional context. Using this attribute displays errors separately and enhances formatting. |

Reviewer 2: ```suggestion
| `error.cause`  | Error | An optional nested error providing additional context. Use this attribute to display errors separately and enhance formatting. |
```
